test/TaskScheduler-test.js

```diff
@@ -121,48 +120,6 @@ describe('TaskScheduler', () => {
     });
   });
 
-  describe('a task scheduler when setImmediate does not exist', () => {
```

Comment:
Maybe keep this test, but repurpose for custom implementation? I like that the test verifies the alternating behavior in another way.

```
describe('a task scheduler with custom setImmediate implementation',
```